ghcup

```diff
@@ -994,7 +994,7 @@ while [ $# -gt 0 ] ; do
            if [ "${TARGET_LOCATION}" ] ; then
                self_update "${TARGET_LOCATION}"
            else
-               self_update "${HOME}/.local/bin"
+               self_update "$(dirname $0)"
```

Comment:
1. this isn't really safe, see https://stackoverflow.com/a/49964642/2976251 and https://stackoverflow.com/questions/59895/getting-the-source-directory-of-a-bash-script-from-within and https://stackoverflow.com/questions/9889938/shell-script-current-directory
2. doesn't pass shellcheck